File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
        }
        answer = map.size() > nums.length / 2 ? nums.length / 2 : map.size();
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    # dict로 빈도수 체크
    dict = {}
    
    for n in nums:
        dict[n] = dict.get(n, 0) + 1
    
    key_num = len(dict)
    
    if len(nums) / 2 <= key_num:
        answer = len(nums) / 2
    else:
        answer = key_num

    return answer
    
    # key 개수와 len(nums) / 2 비교했을때 key가 작거나 작으면 key가 답. key가 더 크면 후자가 답 